# PA-X Peace Agreement Group Project - DS4D Assignment 2

Before we begin any analysis, import all peace agreement data using pandas.

In [8]:
import pandas as pd

pax_all_agreement_data = pd.read_csv('pax_all_agreements_data.csv')

## Describe

- what is the general type of the data (tabular, network, geographical, textual etc.),
- how large and complex is it (rows/columns, size, variation, structure)
- What fields and data types are present (max/min, levels for categorical values).
- Links between this data and other data (e.g. foreign keys, unique ids)
- Summary statistics about the data - how many people, what time frame, field
averages etc.
- How does the data relate to the questions that the data owner has discussed with
you?

The peace agreement database (also known as PA-X Database) provides a downloadable list of all agreements in .csv or 
.xslx format.

In [9]:
all_agreement_data_dimension = pax_all_agreement_data.shape
print("rows: {}, columns: {}".format(all_agreement_data_dimension[0], all_agreement_data_dimension[1]))

rows: 1789, columns: 266


<!-- How large and complex is it (rows/columns, size, variation, structure) -->
The database has record 1789 peace agreements in total and each record comes with 266 columns.

<!-- What fields and data types are present (max/min, levels for categorical values) -->

<!-- Links between this data and other data (e.g. foreign keys, unique ids)-->

<!-- Summary statistics about the data - how many people, what time frame, field
averages etc. -->

<!-- How does the data relate to the questions that the data owner has discussed with
you? -->

## Explore: carry out a deeper exploration of the data. This includes looking at individual
fields/variables to see the distribution of values they take (e.g. evenly distributed, bell curves, bi-modal) or how they are distributed in time. It also includes relationships between variables in your dataset: are there correlations? In which direction? Complex curves? We would expect to see roughly:
- 4-5 exploratory visualisations, presented in a readable form, with an explanation about what you have found
- 1-2 relationships between variables analysed
- Ideas about trends, outliers, clusters
- Reference to statistics, i.e. a sense of which relationships are significant, and what
claims you can back up.

## Reflect and Hypothesise: What do you think might be behind the relationships and
distributions that you have found? How does your data relate to the world? In short, what hypotheses came up as you explored the data, and how would you go about testing them? We would expect to see:
- A 200 word reflection on the data
- 3-5 hypotheses with an explanation of:
  - What your hypothesis is
  - why you think that (referring to your graphs)
  - How you could go about testing it (note that this may include: further studies,

asking the data owner, using some advanced algorithms, checking each data point individually, etc..)
    
**NOTE**: You don’t have to actually test the hypotheses - you just have to demonstrate that you have plausibly thought about how you would test them.